internal/konnect/node_agent.go

```diff
@@ -60,6 +79,10 @@ func (a *NodeAgent) clearOutdatedNodes() error {
 	}
 
 	for _, node := range nodes.Items {
+		// REVIEW: what should the condition be to delete the node in Konnect RG?
+		// (1) Do we check the "last update" of the node, and only delete it when the last update is too old(say, 5 mins ago)?
+		// (2) What if there is a node with the same name but not the same node exists?
+		// for example, When KIC runs in minikube/kind env and whole cluster is stopped then started again.
 		if node.Type == NodeTypeIngressController && node.Hostname != a.Hostname {
```

Comment:
> (1) Do we check the "last update" of the node, and only delete it when the last update is too old(say, 5 mins ago)?
> > (2) What if there is a node with the same name but not the same node exists?
> for example, When KIC runs in minikube/kind env and whole cluster is stopped then started again.

In the case of the KIC node, there should always be only one, so I believe that we may safely remove any existing `NodeTypeIngressController` if it doesn't match our identification despite how much time has passed since its last update.

The second question boils down to defining what identifies the KIC node in our system. IMO this should be tied to a Pod unique identifier and [its lifecycle](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-lifetime). So given we have a Pod rescheduled, the corresponding KIC node in Konnect should be removed and recreated.

I understand that may be challenging now as we do not have API to pass our UID when creating a node, right?

Reply:
done